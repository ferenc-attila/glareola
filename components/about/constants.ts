export const ABOUT = {
    app: {
        title: 'About Glareola Application',
        description: [
            'Read rarity observations from birding.hu',
        ],
        authors: [
            'Attila Ferenc',
        ],
        version: '0.0.1',
        date: '2023-07-15',
        license: {
            shortName: 'MIT',
            link: 'https://opensource.org/license/mit/',
        },
    },
    mainImage: {
        title: 'Collared pratincole (Glareola pratincola)',
        author: 'Saswat Mishra',
        date: '2018-03-30',
        location: 'Nal Sarovar, Gujarat',
        license: {
            shortName: 'CC-BY-SA-4.0',
            link: 'https://creativecommons.org/licenses/by-sa/4.0/',
        }
    }
}
