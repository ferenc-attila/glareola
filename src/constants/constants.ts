export const SOURCES = {
    BIRDING_HU: 'birding.hu',
    BIRDING_SK: 'birding.sk',
    RAREBIRDS_HU: 'rarebirds.hu',
};

export const URLS = {
    BIRDING_HU_MAIN: 'http://birding.hu/index.php?page=&cid=az_elmult_14_nap_ritkasagai&per_page=100',
    BIRDING_HU_GALLERY: 'http://birding.hu/index.php?page=galeriak&cid=ritka_madarfajok&per_page=100',
};

export const URL_TYPES = {
    DATA: 'data',
    GALLERY: 'gallery',
};

export const BUTTON_PROPS = {
    SIZE: 32,
    COLOR: 'white',
};

export const ICON_NAMES = {
    HOME: 'home',
    OBSERVATIONS: 'binoculars',
    INFO: 'info-circle',
    SETTINGS: 'tools',
    IMAGE: 'camera',
    BACK: 'backward',
    MAP: 'map-marked-alt',
    ZOOM: 'search-plus',
    GALLERY: 'images',
};
