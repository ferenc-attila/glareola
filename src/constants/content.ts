export const ABOUT = {
    APP: {
        VERSION: '0.0.7',
        DATE: '2023-10-08',
        LICENSE: {
            SHORT_NAME: 'MIT',
            LINK: 'https://opensource.org/license/mit/',
        },
    },
    COVER_IMAGE: {
        SOURCE: 'https://upload.wikimedia.org/wikipedia/commons/4/4c/Collared_Pratincole_SM.jpg',
        LICENSE: {
            SHORT_NAME: 'CC-BY-SA-4.0',
            LINK: 'https://creativecommons.org/licenses/by-sa/4.0/',
        },
    },
    ICON_IMAGE: {
        SOURCE: 'https://upload.wikimedia.org/wikipedia/commons/9/91/Glareola_pratincola_in_Ambos%C3%A9lie_National_Park_kenya%2C_crop2.jpg',
        LICENSE: {
            SHORT_NAME: 'CC-BY-SA-3.0',
            LINK: 'https://creativecommons.org/licenses/by-sa/3.0/',
        },
    },
};
