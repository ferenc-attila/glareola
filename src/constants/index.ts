export { ABOUT } from './content';
export { SOURCES, URLS } from './constants';
