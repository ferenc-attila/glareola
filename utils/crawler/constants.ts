export const sources = {
    birdingHU: 'birding.hu',
    birdingSK: 'birding.sk',
    rareBirdsHU: 'rarebirds.hu',
}
