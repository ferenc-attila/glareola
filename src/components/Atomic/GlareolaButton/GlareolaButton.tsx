import { FontAwesome5 } from '@expo/vector-icons';
import React from 'react';
import { View, Pressable, StyleSheet } from 'react-native';

import { BUTTON_PROPS } from '../../../constants';
import { COLORS, SPACING } from '../../../styles';
import { IGlareolaButton } from '../../../types/interfaces';

export const GlareolaButton = (props: IGlareolaButton) => {
    return (
        <View style={styles.button}>
            <Pressable accessibilityRole='button' accessibilityLabel={props.accessibilityLabel} onPress={props.onPress}>
                <FontAwesome5
                    name={props.iconName}
                    size={props.size ?? BUTTON_PROPS.SIZE}
                    color={props.color ?? BUTTON_PROPS.COLOR}
                />
            </Pressable>
        </View>
    );
};

const styles = StyleSheet.create({
    button: {
        margin: SPACING.SMALL,
        padding: SPACING.MEDIUM,
        borderRadius: SPACING.SMALL,
        backgroundColor: COLORS.BACKGROUND.BUTTON,
        color: COLORS.FONT.NORMAL,
    },
});
