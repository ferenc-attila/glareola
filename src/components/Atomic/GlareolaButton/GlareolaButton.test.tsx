import { render, screen } from '@testing-library/react-native';
import { fireEvent } from '@testing-library/react-native/build/pure';

import { GlareolaButton } from './GlareolaButton';
import { ICON_NAMES } from '../../../constants';

describe('GlareolaButton', () => {
    test('renders correctly', () => {
        const button = render(
            <GlareolaButton
                onPress={() => {}}
                iconName={ICON_NAMES.HOME}
                size={24}
                color='white'
                accessibilityLabel='testLabel'
            />,
        );
        expect(button.toJSON()).toMatchSnapshot();
    });

    test('calls onPress function when button pressed', async () => {
        const onPress = jest.fn();
        render(<GlareolaButton onPress={onPress} iconName={ICON_NAMES.HOME} accessibilityLabel='testLabel' />);
        fireEvent.press(screen.getByRole('button'));
        expect(onPress).toHaveBeenCalledTimes(1);
    });
});
