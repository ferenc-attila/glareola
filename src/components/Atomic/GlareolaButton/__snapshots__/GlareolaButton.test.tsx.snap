// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlareolaButton renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#4d9460",
      "borderRadius": 5,
      "color": "#fff",
      "margin": 5,
      "padding": 10,
    }
  }
>
  <View
    accessibilityLabel="testLabel"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text />
  </View>
</View>
`;
